fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/FDBLuceneIndexFailureTest.java

```diff
@@ -153,51 +138,55 @@ public static Stream<Arguments> legacySyncAndMapping() {
     @ParameterizedTest
     @MethodSource("legacySyncAndMapping")
     void repartitionGroupedTestWithExceptionMapping(boolean useLegacyAsyncToSync, boolean useExceptionMapping) throws IOException {
-        Index index = COMPLEX_PARTITIONED;
-        Tuple groupingKey = Tuple.from(1L);
         final RecordLayerPropertyStorage contextProps = RecordLayerPropertyStorage.newBuilder()
                 .addProp(LuceneRecordContextProperties.LUCENE_REPARTITION_DOCUMENT_COUNT, 6)
                 .addProp(LuceneRecordContextProperties.LUCENE_USE_LEGACY_ASYNC_TO_SYNC, useLegacyAsyncToSync)
                 .build();
-        setupExceptionMapping(useExceptionMapping);
 
+        final long seed = 6474737L;
+        Tuple groupingKey = Tuple.from(1L);
         final int totalDocCount = 20;
-        Consumer<FDBRecordContext> schemaSetup = context -> rebuildIndexMetaData(context, COMPLEX_DOC, index);
-        long docGroupFieldValue = groupingKey.isEmpty() ? 0L : groupingKey.getLong(0);
+        final LuceneIndexTestDataModel dataModel = new LuceneIndexTestDataModel.Builder(seed, this::getStoreBuilderWithRegistry, pathManager)
+                .setIsGrouped(true)
+                .setPartitionHighWatermark(10)
+                .build();
 
         // create/save documents
+        long docGroupFieldValue = groupingKey.isEmpty() ? 0L : groupingKey.getLong(0);
```

Comment:
`groupingKey` is always `Tuple.from(1L)`, would it make sense to reorder the assignments:
```java
long docGroupFieldValue = 1L;
Tuple groupingKey = Tuple.from(docGroupFieldValue);
```